fairlearn/reductions/resampling.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+import logging
+import math
+# import numpy as np
+# import pandas as pd
+# import scipy.optimize as opt
+# from sklearn import clone
+# from sklearn.dummy import DummyClassifier
+# from time import time
+# import inspect
+# import resampling
+
+
+logger = logging.getLogger(__name__)
+
+
+def resample_fixed_size_deterministic(X, y, weights):
+    """ Resampling methods: Deterministic and keep the sample size.
+
+        Parameters
+        ----------
+        X : {numpy.ndarray, pandas.DataFrame}
+            the training features
+        sensitive_features : {numpy.ndarray, pandas.Series, pandas.DataFrame, list}
+            the sensitive features to use for constraints
+        y : {numpy.ndarray, pandas.Series, pandas.DataFrame, list}
+            the training labels
+        weight : {numpy.ndarray, pandas.Series, pandas.DataFrame, list}
+            the resampling weights
+        """
+
+    factor = len(X) / sum(weights)
+    resampled_X, resampled_y = [], []
+
+    scaled_weight = [weight * factor for weight in weights]
+    for i in range(len(X)):
+        if scaled_weight[i] >= 1 and len(resampled_X) < len(X):
```

Comment:
Doesn't this `if` statement need an `else` ? I'm trying to figure out what's going to happen if there's an extremely 'heavy' element (or two) on the end of the array. Can `add_num` end up causing the size of `resampled_X` to exceed `X` ?